securedrop/tests/test_journalist.py

```diff
@@ -41,6 +42,43 @@ def setUp(self):
     def tearDown(self):
         utils.env.teardown()
 
+    @patch('journalist.app.logger.error')
+    def test_reply_error_logging(self, mocked_error_logger):
+        source, _ = utils.db_helper.init_source()
+        sid = source.filesystem_id
+        self._login_user()
+
+        exception_class = StaleDataError
+        exception_msg = 'Potentially sensitive content!'
+
+        with patch('db.db_session.commit',
+                   side_effect=exception_class(exception_msg)):
+            resp = self.client.post(url_for('reply'),
+                                    data={'sid': sid, 'msg': '_'})
+
+        # Notice the "potentially sensitive" exception_msg is not present in
+        # the log event.
+        mocked_error_logger.assert_called_once_with(
+            "Reply from '{}' (id {}) failed: {}!".format(self.user.username,
+                                                         self.user.id,
+                                                         exception_class))
+
+    @patch('journalist.app.logger.error')
+    def test_reply_error_flashed_message(self, mocked_error_logger):
```

Comment:
Don't mock `journalist.app.logger.error` if you aren't going to use the mock anywhere